protocol/contracts/amo/Ramos.sol

```diff
@@ -453,15 +453,15 @@ contract Ramos is IRamos, TempleElevatedAccess, Pausable {
         (uint256 protocolTokenAmount, uint256 quoteTokenAmount) = protocolTokenBalancerPoolIndex == 0
             ? (request.maxAmountsIn[0], request.maxAmountsIn[1])
             : (request.maxAmountsIn[1], request.maxAmountsIn[0]);
-        protocolTokenVault.borrowProtocolToken(protocolTokenAmount, address(this));
+        tokenVault.borrowProtocolToken(protocolTokenAmount, address(this));
 
         // safe allowance quoteToken and protocolToken
         {
             protocolToken.safeIncreaseAllowance(address(balancerVault), protocolTokenAmount);
             uint256 quoteTokenAllowance = quoteToken.allowance(address(this), address(balancerVault));
+            quoteToken.safeIncreaseAllowance(address(balancerVault), quoteTokenAmount);
             if (quoteTokenAllowance < quoteTokenAmount) {
-                quoteToken.safeApprove(address(balancerVault), 0);
-                quoteToken.safeIncreaseAllowance(address(balancerVault), quoteTokenAmount);
```

Comment:
Hmm why have you moved the increaseAllowance to above the if statement?

This is granting the allowance to `balancerVault`, it doesn't have anything to do with the token vault at this point. When we add liquidity, balancer needs to pull the tokens from Ramos. So this needs to be be put back as it was.

The pattern of first setting to 0 and then setting to an amount is because some tokens (USDC, USDT) don't follow the standard correctly. So we can't just increase the allowance - we need to set to zero and then increase the allowance